### PR TITLE
KHWEBA-5, bar chart

### PR DESCRIPTION
**Pull request by yuriy-dezhniuk**
bar char

### CODE REVIEW

**Comment by gift-a** [comment]
it doesn't work for me.
I do:
npm i
npm start
go to http://localhost:8080/
and see "hi, Ihor!"

UPD now I see that you didn't mount it in app. and it seems that you will mount it inside main component

File: src/components/barChart/barChart.html [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<script src="https://cdnjs.cloudflare.com/ajax/libs/Chart.js/2.4.0/Chart.min.js"></script>
```
Comment (gift-a) [inline src/components/barChart/barChart.html]: as I remember, you should have installed it via npm. I can be mistaken

**Review by gift-a — DISMISSED** [review]

**Comment by Vladyslav-Plakhuta** [comment]
Compiled successfully but I cannot see anything with the exception of(screenshot):

fireshot capture 19 - buxy - http___localhost_8080_

File: src/components/barChart/barChart.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
+"use strict";
```
Comment (ihor7016) [inline src/components/barChart/barChart.js:1]: you do not need use strict  -- if you use es5 modules -- use string is by default

File: src/components/barChart/barChart.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+			this._canvasID = canvasID;
+		}
+		
+		getSetData(data) {
+			if(arguments.length == 0) {
+				return this._data;
+			} else {
+				this._data = data;
+			}
+
```
Comment (ihor7016) [inline src/components/barChart/barChart.js:17]: why empty space here?

File: src/components/barChart/barChart.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+							}
+						}]
+					}
+				}
+			});
+			return myChart;
+		}
+	}
+	
+	//let BCh = new BarChart([10, 34], ["Spent", "Income"], "myChart");
```
Comment (ihor7016) [inline src/components/barChart/barChart.js:71]: why comments here?

File: src/components/barChart/barChart.html [hunk 5]
```diff
@@ -0,0 +1,4 @@
+<div style="width: 500px; height: 450px;">
+    <canvas id="myChart"></canvas>
```
Comment (ihor7016) [inline src/components/barChart/barChart.html:2]: do not use ids -- only classes

File: src/components/barChart/barChart.html [hunk 6]
```diff
@@ -0,0 +1,4 @@
+<div style="width: 500px; height: 450px;">
```
Comment (ihor7016) [inline src/components/barChart/barChart.html:1]: do not use inline styles

File: src/components/barChart/barChart.html (hunk 5 above)
Comment (ihor7016) [inline src/components/barChart/barChart.html:2]: why 4 spaces indent here?

File: src/components/barChart/barChart.js [hunk 7]
```diff
@@ -0,0 +1,73 @@
+"use strict";
+import template from "./barChart.html";
+
+	export class BarChart {
+		constructor(data, labels, canvasID) {
```
Comment (ihor7016) [inline src/components/barChart/barChart.js:5]: please make component constructor arguments look like I showed -- constructor(mountPoint, props)

File: src/components/barChart/barChart.js [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+			this._data = data;
+			this._labels = labels;
+			this._canvasID = canvasID;
+		}
+		
+		getSetData(data) {
```
Comment (ihor7016) [inline src/components/barChart/barChart.js:11]: why do you need this method?

File: src/components/barChart/barChart.html (hunk 1 above)
Comment (ihor7016) [reply src/components/barChart/barChart.html]: ++

**Pull request closed without merging**